src/stdune/path.ml

```diff
@@ -1,5 +1,34 @@
 module Sys = Stdlib.Sys
 
+module Mkdir_p (P : sig
```

Comment:
I wonder if we should just have `mkdir_p` take an `Fpath` (or string, or whatever we decide to call that type). I mean if we have to use `to_string` anyway, why not just use it once?

Reply:
It wouldn't be quite the same. Since `is_root` should terminate depending on the kind of path we have.